src/helm/benchmark/model_deployment_registry.py

```diff
@@ -97,12 +119,148 @@ def register_model_deployments_from_path(path: str) -> None:
         register_model_deployment(model_deployment)
 
 
-def maybe_register_model_deployments_from_base_path(base_path: str) -> None:
-    """Register model deployments from prod_env/model_deployments.yaml"""
-    path = os.path.join(base_path, MODEL_DEPLOYMENTS_FILE)
+def maybe_register_model_deployments_from_base_path(path: str) -> None:
+    """Register model deployments from yaml file if the path exists."""
     if os.path.exists(path):
         register_model_deployments_from_path(path)
 
 
-def get_model_deployment(name: str) -> Optional[ModelDeployment]:
-    return _name_to_model_deployment.get(name)
+# ===================== UTIL FUNCTIONS ==================== #
+def get_model_deployment(name: str) -> ModelDeployment:
+    register_deployments_if_not_already_registered()
+    if name not in DEPLOYMENT_NAME_TO_MODEL_DEPLOYMENT:
+        raise ValueError(f"Model deployment {name} not found")
+    deployment: ModelDeployment = DEPLOYMENT_NAME_TO_MODEL_DEPLOYMENT[name]
+    if deployment.deprecated:
+        hlog(f"WARNING: Model deployment {name} is deprecated")
```

Comment:
We should rethink this. If I run `helm-run --run-specs boolq:model=text --suite deployment --max-eval-instances 5 --models openai/text-davinci-002` I now get >100 lines of deprecation warnings even though I am _not_ using those model deployments.

```
  WARNING: Model deployment ai21/j1-jumbo is deprecated
  WARNING: Model deployment ai21/j1-large is deprecated
  WARNING: Model deployment ai21/j1-grande is deprecated
  WARNING: Model deployment ai21/j1-grande-v2-beta is deprecated
  WARNING: Model deployment anthropic/stanford-online-all-v4-s3 is deprecated
  WARNING: Model name is deprecated. Please use the model deployment name instead.
  Available model deployments for model bigscience/bloom: ['together/bloom']
  WARNING: All model deployments for model bigscience/bloom are deprecated.
```